src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -371,6 +372,10 @@ public void receiveConnection(Socket sock) {
                 connectOne(sid);
             }
 
+        }else if (sid == self.getId()) {
+            closeSocket(sock);
+            throw new ConfigException(String.format("Appearing duplicate SID: %s", sid));
```

Comment:
I think this message could be more descriptive or reshaped. English is not my 1st language, but what about _"Inconsistent ensemble setup: duplicate SID %s found"?_